libr/util/print.c

```diff
@@ -371,13 +371,66 @@ R_API void r_print_byte(RPrint *p, const char *fmt, int idx, ut8 ch) {
 	r_print_cursor (p, idx, 0);
 }
 
+static void print_c_code(RPrint *p, ut64 addr, ut8 *buf, int len, int ws, int w) {
+	len /= ws;
+
+	p->cb_printf ("#define _BUFFER_SIZE %d\n", len);
+
+	switch (ws) {
+	case 2:
+		p->cb_printf ("uint16_t buffer[%d] = {", len);
+		break;
+	case 4:
+		p->cb_printf ("uint32_t buffer[%d] = {", len);
+		break;
+	case 8:
+		p->cb_printf ("uint64_t buffer[%d] = {", len);
+		break;
+	default:
+		p->cb_printf ("uint8_t buffer[%d] = {", len);
+		break;
+	}
+
+	p->interrupt = 0;
+
+	for (int i = 0; !p->interrupt && i < len; i++) {
+		if (!(i % w)) p->cb_printf ("\n  ");
+
+		r_print_cursor (p, i, 1);
+
+		switch (ws) {
+		case 2:
+			p->cb_printf ("0x%04x,", r_read_ble16(buf, p->big_endian));
+			break;
+		case 4:
+			p->cb_printf ("0x%08xU,", r_read_ble32(buf, p->big_endian));
+			break;
+		case 8:
+			p->cb_printf ("0x%016"PFMT64x"ULL,", r_read_ble64(buf, p->big_endian));
+			break;
+		default:
+			p->cb_printf ("0x%02x,", *buf);
+			break;
+		}
+
+		if ((i + 1) < len && (i + 1) % w)
```

Comment:
yes. read the “goto fail” thread fmi. or the modern perl guidelines. both have good reasons to use always brackets even for one line conditionals

> On 31 May 2016, at 11:50, Karol Harasim <notifications@github.com> wrote:
> 
> In libr/util/print.c <https://github.com/radare/radare2/pull/5047#discussion_r65152609>:
> 
> > +		switch (ws) {
> > +		case 2:
> > +			p->cb_printf ("0x%04x,", r_read_ble16(buf, p->big_endian));
> > +			break;
> > +		case 4:
> > +			p->cb_printf ("0x%08xU,", r_read_ble32(buf, p->big_endian));
> > +			break;
> > +		case 8:
> > +			p->cb_printf ("0x%016"PFMT64x"ULL,", r_read_ble64(buf, p->big_endian));
> > +			break;
> > +		default:
> > +			p->cb_printf ("0x%02x,", *buf);
> > +			break;
> > +		}
> > +
> > +		if ((i + 1) < len && (i + 1) % w)
> {} for one line 'if'?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/radare/radare2/pull/5047/files/7e0397fd13ef8da2a4f2dc1a53952729c443bc7b#r65152609>, or mute the thread <https://github.com/notifications/unsubscribe/AA3-li8_kHFbJsjWfCwjQe36AORnXDA-ks5qHAR0gaJpZM4IqA9b>.
>